html/includes/common/top-devices.inc.php

```diff
@@ -85,46 +144,113 @@
+        $table_header = 'CPU Load';
+        if (is_admin() || is_read()) {
+            $query = 'SELECT `hostname`, `last_polled`, `d`.`device_id`, avg(`processor_usage`) as `cpuload` 
+                      FROM `processors` AS `procs`, `devices` AS `d` 
+                      WHERE `d`.`device_id` = `procs`.`device_id` 
+                      AND unix_timestamp() - UNIX_TIMESTAMP(`last_polled`) < :interval 
+                      GROUP BY `d`.`device_id` 
+                      ORDER BY `cpuload` DESC 
+                      LIMIT :count';
+        } else {
+            $query = 'SELECT `hostname`, `last_polled`, `d`.`device_id`, avg(`processor_usage`) as `cpuload` 
+                      FROM `processors` AS procs, `devices` AS `d`, `devices_perms` AS `P`
+					  WHERE `P`.`user_id` = :user AND `P`.`device_id` = `procs`.`device_id` 
+                      AND unix_timestamp() - UNIX_TIMESTAMP(`last_polled`) < :interval
+                      GROUP BY `procs`.`device_id` 
+                      ORDER BY `cpuload` DESC
+                      LIMIT :count';
+        }
+    } elseif ($top_query == 4) {
+        $table_header = 'Memory usage';
+        if (is_admin() || is_read()) {
+            $query = 'SELECT `hostname`, `last_polled`, `d`.`device_id`, `mempool_perc` 
+                      FROM `mempools` as `mem`, `devices` as `d`
+                      WHERE `d`.`device_id` = `mem`.`device_id`
+                      AND `mempool_descr` IN (\'Physical memory\',\'Memory\')
+                      AND unix_timestamp() - UNIX_TIMESTAMP(`last_polled`) < :interval 
+                      ORDER BY `mempool_perc` DESC
+                      LIMIT :count';
+        } else {
+            $query = 'SELECT `hostname`, `last_polled`, `d`.`device_id`, `mempool_perc` 
+                      FROM `mempools` as `mem`, `devices` as `d`, `devices_perms` AS `P`
+                      WHERE `P`.`user_id` = :user AND `P`.`device_id` = `mem`.`device_id`
+                      AND `mempool_descr` IN (\'Physical memory\',\'Memory\')
+                      AND unix_timestamp() - UNIX_TIMESTAMP(`last_polled`) < :interval 
+                      ORDER BY `mempool_perc` DESC
+                      LIMIT :count';
+        }
     }
+
     $common_output[] = '
-<div class="table-responsive">
-<table class="table table-hover table-condensed table-striped bootgrid-table">
-  <thead>
-    <tr>
-      <th class="text-left">Device</th>
-      <th class="text-left">Total traffic</a></th>
-   </tr>
-  </thead>
-  <tbody>
-    ';
+    <div class="table-responsive">
+        <table class="table table-hover table-condensed table-striped bootgrid-table">
+        <thead>
+            <tr>
+                <th class="text-left">Device</th>
+                <th class="text-left">' . $table_header . '</a></th>
+            </tr>
+        </thead>
+        <tbody>';
 
     foreach (dbFetchRows($query, $params) as $result) {
+        if ($top_query == 0) {
```

Comment:
When the variables are updated, you can use === 'something' to be a strict check.